atd-vzv/src/views/nav/SideMapControl.js

```diff
@@ -0,0 +1,121 @@
+import React from "react";
+import { StoreContext } from "../../utils/store";
+
+import styled from "styled-components";
+import { colors } from "../../constants/colors";
+import { ButtonGroup, Button, Card, Label } from "reactstrap";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import {
+  faWalking,
+  faBiking,
+  faCar,
+  faMotorcycle
+} from "@fortawesome/free-solid-svg-icons";
+
+const modeParameters = {
+  pedestrian: {
+    icon: faWalking,
+    syntax: `pedestrian_fl = "Y"`,
+    type: `where`,
+    operator: `OR`
+  },
+  pedalcyclist: {
+    icon: faBiking,
+    syntax: `pedalcyclist_fl = "Y"`,
+    type: `where`,
+    operator: `OR`
+  },
+  motor: {
+    icon: faCar,
+    syntax: `motor_vehicle_fl = "Y"`,
+    type: `where`,
+    operator: `OR`
+  },
+  motorcycle: {
+    icon: faMotorcycle,
+    syntax: `motorcycle_fl = "Y"`,
+    type: `where`,
+    operator: `OR`
+  }
+};
+
+const SideMapControl = () => {
+  const StyledCard = styled.div`
+    font-size: 1.2em;
+
+    .card-title {
+      font-weight: bold;
+      color: ${colors.white};
+    }
+
+    .card-body {
+      background-color: ${colors.white};
+    }
+  `;
+
+  const {
+    mapFilters: [filters, setFilters]
+  } = React.useContext(StoreContext);
+
+  const handleFilterClick = event => {
+    // Set filter or remove if already set
+    const filterName = event.currentTarget.id;
+
+    if (isFilterSet(filterName)) {
+      const filterToRemove = modeParameters[filterName];
+      const updatedFiltersArray = filters.filter(
+        setFilter => setFilter !== filterToRemove
+      );
+      setFilters(updatedFiltersArray);
+    } else {
+      const filter = modeParameters[filterName];
+      const filtersArray = [...filters, filter];
+      setFilters(filtersArray);
+    }
+  };
+
+  const handleAllFiltersClick = () => {
```

Comment:
I was kind of expecting that clicking "All" would visualize highlight all of the modes and the "All" button as well. Maybe that complicates the logic but its essentially the same crashes displayed as when you do this right?

![Screen Shot 2020-01-10 at 9 25 51 AM](https://user-images.githubusercontent.com/5697474/72164490-62d7c500-338b-11ea-8e61-70fcdd9cdfd2.png)

Reply:
That makes sense to me, but the way the logic works here is "All" includes the four modes selected above plus any other modes (train, etc.). It was definitely tough to figure out how this should work with the possibilities of using `OR` or `AND`. 

Right now, the above selection gives result that have pedestrian OR bicycle OR motor vehicle OR motorcycle. I'm open to any changes, and I don't think this will be that difficult to modify.